### DISCUSSION TITLE
ADO pipeline breaking when using 'buildx' plugin to build image

### DISCUSSION THREAD

**Discussion by AsapHogFtw** [docker/buildx#1837] [Q&A]
I am looking for any input to an issue I am experiencing in my ADO pipeline. Below is an example of the first part of the Dockerfile (all that is needed for demonstration) that keeps failing:

    FROM harbor.xyz.biz/foo/bar/sdk:6.0-alpine AS publish
    ENV IS_DOCKER=true
    WORKDIR /src

    COPY . ./
    WORKDIR /src/SpecChannelAirport.Database.Deploy

    RUN dotnet restore --configfile ../NuGet.config
    RUN dotnet build --no-restore
    RUN dotnet publish --no-build -o /output

Now when I set the ENV_VAR DOCKER_BUILDKIT=0 then the pipeline will build the image just fine and without issue; this is because it is using the legacy docker build, which is technically deprecated functionality. However, when this value is set to 1, which it is by default, then the docker-buildx is used, and it causes a very generic error to occur on the last line of the Dockerfile:

     > [7/7] RUN dotnet publish:
    ------
    Dockerfile:10
    --------------------
       8 |     RUN dotnet restore --configfile ../NuGet.config
       9 |     RUN dotnet build --no-restore
      10 | >>> RUN dotnet publish --no-build -o /output
      11 |     
      12 |     # FROM harbor.xyz.biz/foo/bar/server:2019
    --------------------
    ERROR: failed to solve: failed to prepare urqxwwwnadtw7fx2jwt6eszan as u720dwirlzaast49cej32ujs1: invalid argument 

For reference, here is the `build.dockerImage.yml` file that I am using to for the pipeline:

    parameters:
      - name: DOCKER_BUILDKIT
        type: number
        default: 1
        values:
         - 0
         - 1

    jobs:
      - job: 'BuildDockerImage'
        pool:
          name: Linux-dev
          demands:
            - agent.name -equals agent-l-cidocker101.wdn.xyz.io
        workspace:
          clean: outputs
        variables:
          DOCKER_BUILDKIT: ${{ parameters.DOCKER_BUILDKIT }}
        steps:
          - checkout: self
            clean: all

          - task: Docker@2
            displayName: 'Docker: build'
            inputs:
              command: build
              Dockerfile: .pipelines/Dockerfile
              buildContext: $(Build.SourcesDirectory)
              arguments: --pull --no-cache --file .pipelines/Dockerfile

I am using buildx-v0.10.4.linux-amd64 plugin (installed at path /usr/local/lib/docker/cli-plugins/) from [github](https://github.com/docker/buildx/releases/tag/v0.10.4), on my build agent which is an Ubuntu 20.04 docker container, with Docker version 23.0.1 and Docker Compose version 2.17.2.

I was not able to find much in the way of this issue online and am not exactly sure what it could be. If I comment out the last line of the Dockerfile, it will build fine, even when using the buildx plugin, so I feel that something has to be going on between the way that the legacy build and the current buildx is interacting or affecting the way that the dotnet publish command is operating.

Does anyone have any ideas or insight to this kind of problem?

**Comment by mkim0203** [comment]
docker build 환경이 docker 컨테이너로 되어 있을경우 동일한 증상이 발생적이 있습니다.
host에 있는 docker sock을 컨테이너로 공유 받아서 빌드를 진행하니 증상이 해결되었습니다.

같은환경이면 host에 docker build 환경을 구성후  컨테이너로 공유하여 진행해보시기 바랍니다.

The same symptom has occurred when the docker build environment is a docker container.
The symptom was resolved by sharing the docker sock in the host as a container and proceeding with the build.

If it is the same environment, please proceed by configuring the docker build environment on the host and sharing it as a container.

**Reply by AsapHogFtw** [reply]
The `/var/run/docker.sock` path is already a mounted volume between the build-agent container and the host machine. I do not think that that is the problem. If you could, please provide me with the link that you think answers this question.
